designs/sam-config.md

```diff
@@ -38,53 +38,55 @@ Tenets
 What will be changed?
 ---------------------
 
-The suite of commands supported by SAM CLI would be aided by looking for a configuration file thats locally located under the `.aws-sam/` at the project root where template.yaml is located by default.
+The suite of commands supported by SAM CLI would be aided by looking for a configuration file thats locally located under at the project root where template.yaml is located by default.
 
-`.aws-sam/samconfig.toml`
+`samconfig.toml`
 
 
 This configuration would be used for specifiying the parameters that each of SAM CLI commands use and would be in TOML format.
 
-Running a SAM CLI command now automatically looks for `.aws-sam/samconfig.toml` file and if its finds it goes ahead with parameter passthroughs to the CLI.
+Running a SAM CLI command now automatically looks for `samconfig.toml` file and if its finds it goes ahead with parameter passthroughs to the CLI.
 
 ```
 sam build
-Default Config file location: .aws-sam/samconfig.toml
+Default Config file location: samconfig.toml
 ..
 ..
 ..
 ```
 
-Why samconfig under `.aws-sam`
+Why `samconfig.toml` not under `.aws-sam` directory?
 ---------------------------------
 
-The `.aws-sam` directory within the project directory is created with normal 755 permissions as default without any special permisions. `sam build` only creates a build directory within `.aws-sam` as `.aws-sam/build`. This directory is erased and re-built on every build. but top level directory is left unaffected.
-
-The `.gitignore` specified in the init apps also only have `.aws-sam/build` ignored and not anything else.
+The `.aws-sam` directory generally holds artifacts that are driven by the implementation details behind the sam cli commands themselves. A configuration file which controls how a sam app behaves should exist by itself as a first class citizen right next to the sam template.
 
 
 Config file versioning
 -----------------------
 
-The configuration file: `samconfig.toml` will come with a top level version key that specifies the version of the configuration file. This version can then be used to determine if a given configuration file works with a version of SAM CLI.
+The configuration file: `samconfig.toml` will come with a top level version key that specifies the version of the configuration file based on the spec of the file. This version can then be used to determine if a given configuration file works with a given version of SAM CLI.
 
 It also paves the forward when major changes need to be made to the configuration file and add a version bump to the config file version
 
 ```
 version = 0.1
 ```
 
+* SAM CLI expects the version format to follow a float format
```

Comment:
you mean semantic version format? 0.1.1 is not a valid float

Reply:
Done